pkg/disk/utils.go

```diff
@@ -0,0 +1,63 @@
+package disk
+
+import (
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	"os"
+	"os/exec"
+)
+
+func DownloadDiskImageFromURL(rawDiskUrl, headerKey, headerValue, certificatePath, diskPath string) error {
+	cmd := exec.Command(
+		"nbdkit",
+		"-r",
+		"curl",
+		rawDiskUrl,
+		fmt.Sprintf("header=%s: %s", headerKey, headerValue),
+		fmt.Sprintf("cainfo=%s", certificatePath),
+		"--run",
+		fmt.Sprintf("qemu-img convert \"$uri\" -O qcow2 %s", diskPath),
+	)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+
+	if fileInfo, err := os.Stat(diskPath); err != nil || fileInfo.Size() == 0 {
+		return fmt.Errorf("disk image file does not exist or is empty")
+	}
+	return nil
+}
+
+func CreateFile(path, data string) error {
```

Comment:
Add this function to vm-export package, since it creates certificate for it.

Reply:
I consider `export.go` as a client to interact with the API server. We can also remove `utils.go` and move all functions there, do you prefer it?